src/tensors/indexmanipulations.jl

```diff
@@ -473,78 +473,156 @@ function add_transform!(tdst::AbstractTensorMap,
     return tdst
 end
 
-function add_transform_kernel!(tdst::TensorMap,
-                               tsrc::TensorMap,
-                               (p₁, p₂)::Index2Tuple,
-                               ::TrivialTreeTransformer,
-                               α::Number,
-                               β::Number,
-                               backend::AbstractBackend...)
-    return TO.tensoradd!(tdst[], tsrc[], (p₁, p₂), false, α, β, backend...)
+function use_threaded_transform(t::TensorMap, transformer::TreeTransformer)
+    return get_num_transformer_threads() > 1 && length(t.data) > Strided.MINTHREADLENGTH
```

Comment:
As this does not use the `transformer` argument, we might choose not to restrict it, such that `use_threaded_transform` can also deal with the case where `transformer` is simply function (i.e. the `AbstractTensorMap` case), and the old/generic implementations are used (which also have some multithreading built in).